### DISCUSSION TITLE
Ontologies for annotating NMR Meta data

### DISCUSSION THREAD

**Discussion by vcnainala** [NFDI4Chem/nmrxiv#95] [Ontologies]
### **Ontologies**
In nmrXiv, we aim to provide the data, present it unambiguously, and make it [FAIRer](https://www.go-fair.org/fair-principles) (Findable, Accessible, Interoperable, Reusable). Our focus is not to make the data just human-readable but machine-readable so researchers can leverage advanced machine learning and other language processing techniques to re-analyse the data. This can be facilitated by using ontologies. 

However, it is challenging to make ontologies exhaustive as new terms arise, especially in emerging domains such as metabolomics, with NMR being the central analytical technique.

Being an NMR data repository, we find ourselves suited for gathering the relevant missing terms via user submission and contributing those missing terms to ontologies we find relevant to nuclear magnetic resonance (NMR) data or repositories data in general (in other words, to domain-specific and domain-independent ontologies). We aim to achieve this through collaboration with the NFDI4Chem Terminology service (Task Area 6).

### **Domain-independent Ontologies:**
The following ontologies provide identifiers, references, formats, and possible operations to define the experiment.

- [Bioinformatics operations, data types, formats, identifiers and topics - EDAM](https://terminology.nfdi4chem.de/ts/ontologies/edam)
- [REPRODUCE-ME Ontology - REPR](https://www.ebi.ac.uk/ols/ontologies/reproduceme)
- [Medical Subject Headings - MeSH](https://meshb-prev.nlm.nih.gov/treeView)
- [NCI Thesaurus OBO Edition - NCIt](https://www.ebi.ac.uk/ols/ontologies/ncit)

### **Domain-specific Ontologies:**

- [nuclear magnetic resonance CV - NMR](https://terminology.nfdi4chem.de/ts/ontologies/nmrcv):
We believe nmrCV is highly relevant for nmrXiv as it covers many aspects of the NMR assay and the resulting spectral data, including the files formats, dimensionality, instruments, used materials, and later data processing.

- [Chemical Methods Ontology - CHMO](https://terminology.nfdi4chem.de/ts/ontologies/chmo):
Methods used to collect data in chemical experiments, including NMR spectroscopy, is covered under this ontology.

- [NCBI organismal classification - NCBITAXON](https://www.ebi.ac.uk/ols/ontologies/ncbitaxon):
Widely used curated classification for organisms.

- [The BRENDA Tissue Ontology - BTO](https://www.ebi.ac.uk/ols/ontologies/bto):
It covers organism parts and variants. Examples include tissues, cell lines, cell types and cell cultures.
- [Experimental Factor Ontology - EFO](https://www.ebi.ac.uk/ols/ontologies/efo):
For many experimental variables, which can be used, e.g., to describe the tissue and the organism part.
- [NCI Thesaurus OBO Edition - NCIt](https://www.ebi.ac.uk/ols/ontologies/ncit):
It covers cells, and other materials and attributes.
- [Medical Subject Headings - MeSH](https://meshb-prev.nlm.nih.gov/treeView):
Medical Subject Headings is a comprehensive controlled vocabulary for indexing journal articles and books in the life sciences. 
- [Ontology for Biomedical Investigations - OBI](https://terminology.nfdi4chem.de/ts/ontologies/obi):
Ontology for Biomedical Investigations (OBI) serves as a resource for annotating biomedical investigations, including the study design, protocols and instrumentation used, the data generated and the types of analysis performed on the data. This ontology arose from the Functional Genomics Investigation Ontology (FuGO) and contains both terms common to all biomedical investigations, including functional genomics investigations and those that are more domain-specific.
- [CHEBI Integrated Role Ontology - CHIRO](https://terminology.nfdi4chem.de/ts/ontologies/chiro):
CHEBI provides a distinct role hierarchy for chemical entities.
- [chemical information ontology - CHEMINF](https://terminology.nfdi4chem.de/ts/ontologies/cheminf):
CHEMINF provides terms for cheminformatics software and algorithms, such as chemical identifiers and others. 
- [Units of measurement ontology - UO](https://terminology.nfdi4chem.de/ts/ontologies/uo):
For metric units.

**Comment by vcnainala** [comment]
@okoepler @hujo91 @StroemPhi 
Can NFDI4Chem terminology service support the ontologies (mentioned above) that are not currently available??

**Comment by tilfischer** [comment]
Further relevant chemistry-specific ontologies are [CHEMINF](https://www.ebi.ac.uk/ols/ontologies/cheminf) and possibly [AFO](https://www.ebi.ac.uk/ols/ontologies/afo).

**Reply by NRayya** [reply]
I think CEMINF can be useful for, e.g.,  Chemical identifiers, but in which context do you suggest the use of AFO?

**Reply by tilfischer** [reply]
It depends on what terms you need and one may find in CHEMINF or the ontologies mentioned above.

**Reply by StroemPhi** [reply]
With regard to reusing AFO we probably need to organize a seperate meeting, as AFO is manually curating its import modules composed of OBO ontologies (RO, IAO, OBI, CHMO). This will most likely lead to problems when one uses AFO modules together with OBO modules. But details regarding this still need to be investigated.

**Comment by okoepler** [comment]
You can check included ontologies here: https://terminology.nfdi4chem.de/ts/ontologies
If a ontology is within the scope of NFDI4Chem we can check / add it to the terminology service. We have defined some criteria like open license, accessibility .. before adding them

**Reply by NRayya** [reply]
This is a list of the ontologies not found in the terminology service, we will update it in the future when we see the need for more ontologies. https://docs.google.com/spreadsheets/d/1xg09q6Ee6RmoCCbTDIHk_w9fFN6QC2x7RqlmYoiNo34/edit#gid=0
The list contains the licenses and the ontology IRI and how it is related to chemistry, the main issue is that most of them are biology related as we need this information for the organism, organism part, and variant

**Comment by NRayya** [comment]
Additionally, I didn't find ontologies providing terms for absolute and relative configuration. I think we will need those terms in nmrXiv.

**Comment by StroemPhi** [comment]
@NRayya can you please specify what you mean by absolute and relative configuration?

**Reply by NRayya** [reply]
Here is an example where the relative configuration of the molecule (trans) is provided:
https://terminology.nfdi4chem.de/ts/ontologies/chebi/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCHEBI_35100
However, I didn't manage to find the term "trans" itself
Terms needed for Chemotion fields

- Relative configuration terms are: cis, trans,, ortho, meta, para,, syn, anti

- Absolute configuration terms are: R, S

More details in the table at the end of the page:
https://en.wikipedia.org/wiki/Absolute_configuration

**Reply by hujo91** [reply]
- regarding "[absolute configuration](https://bioportal.bioontology.org/ontologies/GLYCO?p=classes&conceptid=http%3A%2F%2Fglycomics.ccrc.uga.edu%2Fontologies%2FGlycO%23absolute_configuration)" this maybe a starting point even if this definition is perhaps a bit very specific to amino acids and sugars
- for "relative configuration" I have unfortunately found nothing so far
- maybe also PATO and the classes "[liquid configuration](https://bioportal.bioontology.org/ontologies/PATO/?p=classes&conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001735)" & "[solid configuration](https://bioportal.bioontology.org/ontologies/PATO/?p=classes&conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001736)" could be of interest

**Comment by NRayya** [comment]
We can use PATO for [qualities ](https://terminology.nfdi4chem.de/ts/ontologies/pato) to describe the NMR assay such as temperature and wavelength..

**Comment by hujo91** [comment]
maybe CHEMINF "[stereochemical descriptor](https://terminology.nfdi4chem.de/ts/ontologies/cheminf/terms?iri=http%3A%2F%2Fsemanticscience.org%2Fresource%2FCHEMINF_000027&viewMode=All&siblings=false)" class can be useful for the description of sterospecific configuration

**Reply by NRayya** [reply]
Yeah it is perfect, thanks.
But when I searched for those terms, I got the molecules suggested first for several pages.
This brings us back again to the ranking issue

**Comment by NRayya** [comment]
I have started collecting the missing terms [here](https://docs.google.com/spreadsheets/d/1YNK22XZ7jAK9z_hv0HO5q6dKIqQX_UFlM_PcXVyhqzQ/edit#gid=0). the file can be found in TA6 workspace

**Comment by NRayya** [comment]
We are facing an issue: when we ask the user to provide a field name as an ontology term, we can have some combinations of an ontology term like "organism" but with words like "number of, duration of, temperature of..." (that can also be described with ontology terms)  so the resulting field name is difficult to be provided as one ontology term. Here are some real examples:

- **duration of** incubation (from Raman dataset)
-  **number of** rings (NMRShiftDB)
- **size of** smallest set of smallest rings (NMRShiftDB)

**Comment by NRayya** [comment]
Further discussions regarding NMR related ontologies will take place in the NMR task force meeting.
